src/main/kotlin/com/openlattice/linking/blocking/ElasticsearchBlocker : Blocker.kt

```diff
@@ -106,7 +106,16 @@ class ElasticsearchBlocker(
              * The main linking service will skip these elements anyway so we should avoid loading data related to
              * pathological cases
              */
-            return entityDataKey to mapOf()
+            /* There can be cases, when there is no sufficient data for an entity to be blocked to itself
+             * (example: only 1 property has value)
+             * If it cannot block to itself, we only link it to itself
+             */
+            return entityDataKey to dataLoader
```

Comment:
can't you just do selfBlock.add(entityDataKey.entityKeyId)? It will automatically be picked up when doing loadedData...I think

Reply:
No, because selfblock is not used to return values, only blockedEntitySetSearchResults.

B option is to empty blockedEntitySetSearchResults and add self / reassign blockedEntitySetSearchResults